Ruby/Ruby.sublime-syntax

```diff
@@ -209,84 +209,65 @@ contexts:
         3: storage.type.numeric.imaginary.ruby
-        4: invalid.illegal.numeric.rational.ruby
+        4: storage.type.numeric.imaginary.ruby
+        5: invalid.illegal.numeric.rational.ruby
     # numbers with binary base prefix
-    - match: '\b(0[bB]){{bdigits}}(r)?(?:(i)(r)?)?\b'
+    - match: '\b(0[bB]){{bdigits}}(r)?(?:(\.i)|(i)(r)?)?\b'
       scope: constant.numeric.integer.binary.ruby
       captures:
         1: punctuation.definition.numeric.binary.ruby
         2: storage.type.numeric.rational.ruby
         3: storage.type.numeric.imaginary.ruby
-        4: invalid.illegal.numeric.rational.ruby
-    # complex imaginary numbers
-    - match: |-
-        \b(?x:
-          {{ddigits}} (?:
-            # 1ri, 1.1ri, 1i, 1.1i | 1e1i, 1.1e1i
-            (?: (\.) {{ddigits}} )? (?: (r) | {{exponent}}? )
-            # 1.ri, 1.e1ri, 1.1e1ri | 1.i 1.e1i
-            | (\.) (?: {{ddigits}}? {{exponent}}? (r) | ({{exponent}})? )
-          )
-        )(i)(r)?\b
+        4: storage.type.numeric.imaginary.ruby
+        5: invalid.illegal.numeric.rational.ruby
+    # complex imaginary numbers: 1i, 1.i, 1.1i, 1.1.i, 1e1i, 1e1.i, 1.1e1i, 1.1e1.i, 1ri, 1r.i, 1.1ri, 1.1r.i
+    - match: '\b{{ddigits}}(?:(\.){{ddigits}})?(?:(r)|{{exponent}}(r)?)?(?:(\.i)|(i)(r)?)\b'
       scope: constant.numeric.complex.imaginary.ruby
       captures:
         1: punctuation.separator.decimal.ruby
         2: storage.type.numeric.rational.ruby
-        3: punctuation.separator.decimal.ruby
-        4: invalid.illegal.numeric.rational.ruby
-        5: invalid.illegal.numeric.exponent.ruby
-        6: storage.type.numeric.imaginary.ruby
-        7: invalid.illegal.numeric.rational.ruby
-    # rational numbers
-    - match: |-
-        \b(?x:
-          # 1.r | 1r, 1.1r, 1.e1r, 1.1e1r
-          {{ddigits}} (?: (\.) (r) | (?: (\.) {{ddigits}}? ({{exponent}})? )? (r) )
-        )\b
+        3: invalid.illegal.numeric.rational.ruby
+        4: storage.type.numeric.imaginary.ruby
+        5: storage.type.numeric.imaginary.ruby
+        6: invalid.illegal.numeric.rational.ruby
+    # rational numbers: 1r, 1.1r
+    - match: '\b{{ddigits}}(?:(\.){{ddigits}})?(r)'
       scope: constant.numeric.rational.ruby
       captures:
         1: punctuation.separator.decimal.ruby
-        2: invalid.illegal.numeric.rational.ruby
-        3: punctuation.separator.decimal.ruby
-        4: invalid.illegal.numeric.exponent.ruby
-        5: storage.type.numeric.rational.ruby
-    # floating point numbers
-    - match: |-
-        \b(?x:
-          {{ddigits}} (?:
-            # 1.1, 1.1e1, 1.e1
-            (\.) (?: {{ddigits}} {{exponent}}? | ({{exponent}}) )
-            # 1e1
-            | {{exponent}}
-          )
-        )\b
+        2: storage.type.numeric.rational.ruby
+    # floating point numbers: 1.1, 1e1, 1.1e1
+    - match: '\b{{ddigits}}(?:(\.){{ddigits}}|(?:(\.){{ddigits}})?{{exponent}}(r)?)\b'
       scope: constant.numeric.float.ruby
```

Comment:
In some syntaxes like R it is possible to have `constant.numeric.float.binary`, ... . Therefore we started adding `.decimal` to normal floats as well: `constant.numeric.float.decimal.ruby`. 

Not sure whether we really need it, but it would cause decimal floats to have the same scope in all languages, if we make it to apply it everywhere.

Reply:
Ok, does it makes sense to update the scope for `0xAAr` to be `constant.numeric.rational.hexadecimal` too then? But as discussed previously, this would require additional rules and I still can't think of an optimal and consistent scope for `0xAAri`.

Reply:
If we consider `rational` as a nother data type next to integer and float,  `constant.numeric.rational.hexadecimal` would definitely make sense in manner of consistency.

I am a bit uncertain about imaginary numbers as this is just another perspective of how to scope numbers, but Go just ignores them. It just handles everything as integer/float/... with the `i` scoped as `storage.type`.

As we concluted scoping complex numbers to result in too complex patterns the Go approach might even be just enough.

But I don't feel like the one to decide it.